shared/keystore/keystore.go

```diff
@@ -0,0 +1,223 @@
+package keystore
```

Comment:
This is from geth right? Let's give credit by adding the header comments geth adds to their files here